gadgetctl/kubectl-gadget: Crash when running `attach` right after `run`

Consider the following manifest:

```yaml
# manifest.yaml
apiVersion: 1
kind: instance-spec
image: trace_exec
name: mygadgetinstance
tags:
  - foo
  - bar
paramValues:
```

```bash
$ sudo gadgetctl run -f manifest.yaml --detach && sudo gadgetctl attach mygadgetinstance
INFO[0000] installed as "729e15bb1655c84211a742c8ff7479a7"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xa0 pc=0x307ccc8]

goroutine 1 [running]:
github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-context.(*GadgetContext).LoadGadgetInfo(0xc00052c820, 0x0, 0xc000130f00, 0x0)
        /home/mauriciov/kinvolk/ebpf/inspektor-gadget/pkg/gadget-context/gadget-context.go:320 +0xc8
github.com/inspektor-gadget/inspektor-gadget/pkg/runtime/grpc.(*Runtime).GetGadgetInfo(0xc0008b4870, {0x46fe7e0, 0xc00052c820}, 0xd?, 0xc000130f00)
        /home/mauriciov/kinvolk/ebpf/inspektor-gadget/pkg/runtime/grpc/oci.go:62 +0x32c
github.com/inspektor-gadget/inspektor-gadget/cmd/common.NewRunCommand.func1(0xc000754908, {0xc0003104c0, 0x1, 0x1})
        /home/mauriciov/kinvolk/ebpf/inspektor-gadget/cmd/common/oci.go:223 +0xc65
github.com/spf13/cobra.(*Command).execute(0xc000754908, {0xc0003104c0, 0x1, 0x1})
        /home/mauriciov/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:970 +0xa44
github.com/spf13/cobra.(*Command).ExecuteC(0xc0004be908)
        /home/mauriciov/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1117 +0x3ff
github.com/spf13/cobra.(*Command).Execute(...)
        /home/mauriciov/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1041
main.main()
        /home/mauriciov/kinvolk/ebpf/inspektor-gadget/cmd/gadgetctl/main.go:108 +0x905
````

Adding a sleep between the commands fixes it:

```bash
$ sudo gadgetctl run -f manifest.yaml --detach && sleep 2 && sudo gadgetctl attach mygadgetinstance
INFO[0000] installed as "77922556b8944cff8b2ace4ed3075e62"
RUNTIME.CONTAINERNAME                                    COMM                    PID        TID ARGS                               ERRO
```

--- a/pkg/gadget-service/instance-manager/instance.go
+++ b/pkg/gadget-service/instance-manager/instance.go
@@ -59,9 +59,11 @@ type GadgetInstance struct {
 	cancel               func()
 	state                gadgetState
 	error                error
+	ready                chan struct{}
 }
 
 func (p *GadgetInstance) GadgetInfo() (*api.GadgetInfo, error) {
+	<-p.ready
 	p.mu.Lock()
 	defer p.mu.Unlock()
 	return p.gadgetInfo, p.error
@@ -186,6 +188,7 @@ func (p *GadgetInstance) Run(
 				Payload: d,
 			}
 			p.gadgetInfo = gi
+			close(p.ready)
 			return nil
 		}),
 	)

--- a/pkg/gadget-service/instance-manager/manager.go
+++ b/pkg/gadget-service/instance-manager/manager.go
@@ -101,6 +101,7 @@ func (m *Manager) RunGadget(instance *api.GadgetInstance) {
 		eventBufferOffs: 0,
 		cancel:          cancel,
 		clients:         map[*GadgetInstanceClient]struct{}{},
+		ready:           make(chan struct{}),
 	}
 	m.mu.Lock()
 	m.gadgetInstances[gi.id] = gi

